File: src/main/java/bad/robot/radiate/teamcity/Password.java
package bad.robot.radiate.teamcity;

@Deprecated
public class Password extends bad.robot.http.configuration.AbstractValueType<String> {

    public static Password password(String password) {
        if (password == null)
            return new NoPassword();
        return new Password(password);
    }

    protected Password(String value) {
        super(value);
    }

    public bad.robot.http.configuration.Password asSimpleHttp() {
        return bad.robot.http.configuration.Password.password(value);
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/Password.scala
package bad.robot.radiate.teamcity

object Password {
  def password(password: String): Password = {
    if (password == null) NoPassword else new Password(password)
  }
}

// todo implicit to convert the SimpleHTTP password
case class Password(value: String) {
  def asSimpleHttp: bad.robot.http.configuration.Password = {
    bad.robot.http.configuration.Password.password(value)
  }
}

object NoPassword extends Password("no password")